File: 3-longestSubstring/Go/longestSubstring_test.go
package longestsubstring

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)


func TestArray(t *testing.T){
	for _, c := range []struct {
		str []string
		want int
	}{
		{[]string{""}, 0},
		{[]string{"a","aa", " ", "111", "?"}, 1}, 
		{[]string{"ab", " 2", "/1", "1/1"}, 2},
		{[]string{"abc", "abcabcbba", "dvds", "abccba"}, 3},
		{[]string{"abcd","abcdabcd", "abcabcd", "1234432111", "^%$?"}, 4}, 
	} {
		for _, s := range c.str{
			got := LengthOfLongestSubstring(s)
			if !cmp.Equal(got, c.want) {
				t.Errorf("LengthOfLongestSubstring(%v) got == %v, want %v", c.str, got, c.want)
			}

		}
	}
	
}

File: 3-longestSubstring/Python3/longest_substring_test.py
''' Given a string s, find the length of the 
    longest substring without repeating characters. '''

import unittest
from longest_substring import Solution

class Test(unittest.TestCase):
    '''testing'''

    def tests(self):
        '''basic tests'''
        in_arrays = ["", "abc", "bb", "dvds", "abcabcbb", "abccbafa"]
        wants = [0, 3, 1, 3, 3, 4]
        for i, s in enumerate(in_arrays):
            got = Solution().length_of_longest_substring(s)
            want = wants[i]
            print(f'Test {i} - got: {got} | want: {want}')
            self.assertEqual(got, want)

if __name__ == '__main__':
    unittest.main()
